addons/test_mail/tests/test_mail_composer.py

```diff
@@ -729,21 +735,30 @@ def test_mail_composer_recipients(self):
             'reply_to': False,
         })
 
-        for composition_mode, batch in (('comment', False), ('mass_mail', True)):
-            with self.subTest(composition_mode=composition_mode, batch=batch):
-                self.assertFalse(
-                    self.env['res.partner'].search([
-                        ('email_normalized', 'in', ['test.cc.1@test.example.com',
-                                                    'test.cc.2@test.example.com'])
-                    ])
-                )
+        for is_manual_main_recipient, (composition_mode, batch) in \
```

Comment:
Wondering if this should not be a specific unit test ... makes the code a bit hard to read (as it is already a bit spaghetti ^^ ) . There is a class "results comment" testing the results of using the composer in comment mode, maybe it could be moved in that class (testing the "final result" of a posting process). This class is more about the composer content itself.

Reply:
Yes I should have made it independent directly because I did rebase that part a few times :-). It is now done.